### PR TITLE
Abstract button class

### PR DESCRIPTION
**Pull request by studioant**
Changes to:
buttons.h - feature now abstract
program.cpp
helper.h - now has functions from program
cdata - add 'this' keyword to property accessors
splashscreen.h - feature now draws type Button
includes.h - order matters

### CODE REVIEW

**Review by sarahgos — APPROVED** [review]
This all looks great to me Anthony. I like the additions to the Button class and removing the functions from program and adding them into helper. Interested to see the Splashscreen and looking forward to trying out these changes! Great work.

File: splashscreen.h [hunk 1]
```diff
@@ -21,27 +25,25 @@ class Splashscreen {
...
 
-            sprite_set_position(exitButton, exitPosition);
-            draw_sprite(exitButton, 0, 0);
+            // Draw text on buttons
+            draw_text("play!", COLOR_BLANCHED_ALMOND, "btn_font", 80, btns[0]->x() - (btns[0]->centre_x()/2) + 5, btns[0]->y() - btns[0]->centre_y(), option_to_screen());
+            draw_text("options", COLOR_BLANCHED_ALMOND, "btn_font", 80, btns[1]->x() - (btns[1]->centre_x()/2) - 20, btns[1]->y() - btns[1]->centre_y(), option_to_screen());
+            draw_text("exit", COLOR_BLANCHED_ALMOND, "btn_font", 80, btns[2]->x() - (btns[2]->centre_x()/2) + 20, btns[2]->y() - btns[2]->centre_y(), option_to_screen());
         }
 
         void button_clicked(point_2d point)
```
Comment (Delcari) [inline splashscreen.h:43]: Consider refactoring to use the button classes, action function. 
(if not already intending to)

File: buttons.h [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        void draw_screen_guides()
+        {
+            draw_circle(COLOR_BLACK, WIDTH*0.5 , HEIGHT*0.5 , 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.25, HEIGHT*0.5 , 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.75, HEIGHT*0.5 , 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.5 , HEIGHT*0.25, 3);
+            draw_circle(COLOR_BLACK, WIDTH*0.5 , HEIGHT*0.75, 3);                                    
+        }
+
+        string btn_color(Color c)
```
Comment (Delcari) [inline buttons.h:103]: Maybe this function should exist within one of the sub-classes, considering we are going to have a number of button assets I assume for menu, options, etc.

**Review by Delcari — APPROVED** [review]
Implementation seems good, works as expected.
Please review aforementioned comments and address if deemed necessary.
Happy to see this merged into the main branch so I can begin adding support within the gridlayout class!
**Great work** Anthony

**Pull request merged**